### DISCUSSION TITLE
Is it possible to get FastApi global state?

### DISCUSSION THREAD

**Discussion by smuglik** [aminalaee/sqladmin#517] [Q&A] [answered]
Hi, 

Is it possible to get FastAPI global state inside ModelView events? 

Thank you in advance!

**Comment by rossmacarthur** [comment, marked as answer]
Also had this issue, I had to attach the FastAPI app to the admin app to be able to access it in requests to the admin app.

```python
app = FastAPI()
admin = Admin(app, engine)
admin.admin.parent = app
```

Now in your request you can access the state like this

```python
request.app.parent.state
```
